requirements.txt

```diff
@@ -1,3 +1,4 @@
+astunparse

```

Comment:
The TensorFlow pip packages should automatically install `astunparse` as a required dependency as listed in their package metadata and same with [setup.py](https://github.com/tensorflow/tensorflow/blob/8faf71f12fdb6f75a06cc4455c19da7e2e1ab4ba/tensorflow/tools/pip_package/setup.py#L86) when installing from source.

```
Metadata-Version: 2.1
Name: tensorflow
Version: 2.12.0
...
Requires-Dist: astunparse (>=1.6.0)
...
```
```
Metadata-Version: 2.1
Name: tensorflow-intel
Version: 2.12.0
...
Requires-Dist: astunparse (>=1.6.0)
...
```

So this seems redundant when the webui base itself doesn't even install TensorFlow, only some extensions do. None the less, if you feel this is really needed, it would be wise to pin the same version range as TensorFlow itself.

Reply:
> The TensorFlow pip packages should automatically install astunparse as a required dependency as listed in their package metadata and same with [setup.py]

yup, but there was a buggy installer in tf 2.11 and 2.12 only came out very recently.

> it would be wise to pin the same version range as TensorFlow itself.

i hate specifying version range where absolutely not needed - last version of `astunparse` is from 2019, its unlikely that a brand new breaking version is going to popup out-of-the-blue.

> So this seems redundant when the webui base itself doesn't even install TensorFlow, only some extensions do. 

very true. i'm just trying to make it as least error prone as possible for average users as i've seen this happen on multiple systems.

> if you feel this is really needed

i don't - i'm just trying to make it error-proof - but i'm open to suggestions  
(and removing astunparse from dependencies if desired).

Reply:
> there was a buggy installer in tf 2.11

If this is the case, then maybe `astunparse` could be moved to the launch.py and be conditional on tensorflow being installed? Something like:

```
    if is_installed("tensorflow") or is_installed("tensorflow-intel") or is_installed("tensorflow-gpu"):
        if not is_installed("astunparse"):
            run_pip(f"install {astunparse}", "astunparse")
```
Though astunparse is a tiny package which hasn't been updated since 2019, so it probably isn't a big deal if it gets mistakenly installed when not needed. I'll leave judgement up to you and automatic, since I have no firm opinion about this.